core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -382,6 +387,65 @@ protected static String getRandomRewriteMethod() {
     }
 
     protected String getRandomType() {
-        return (currentTypes.length == 0) ? MetaData.ALL : randomFrom(currentTypes);  
+        return (currentTypes.length == 0) ? MetaData.ALL : randomFrom(currentTypes);
+    }
+
+    /**
+     * Helper method to return a random field (mapped or unmapped) and a value
+     */
+    protected static Tuple<String, Object> getRandomFieldNameAndValue() {
```

Comment:
potentially but I don't see why we add a method in the base class if we don't use it everywhere. then add it to the class that needs it only?